python/ros_interface.py

```diff
@@ -0,0 +1,107 @@
+import rospy
+import numpy
+from sailbot_msg.msg import Sensors, windSensor, GPS
+
+# Constants
+CHECK_PERIOD_SECONDS = 0.1  # How often fields are updated
+KNOTS_TO_KMPH = 1.852
+ERROR = 0.2  # If signals need different margin of errors, modify as required
+
+# Conversion Notes:
+# - wind_speedKmph, wind_direction, wind_headingDegrees, gps_speedKmph: int32 -> float64 (no loss of precision)
+# - wind_speedKmph, gps_speedKmph: knots -> km/h
+# - wind_direction: no conversion required (0 is bow, 90 is starboard/right, etc - cw - degrees)
+# - heading_degrees: (0 is north, 90 is east, etc - cw - degrees) -> (0 is east, 90 is north, etc - ccw - degrees)
+
+
+class RosInterface:
+    def __init__(self):
+        rospy.init_node('ros_interface', anonymous=True)
+        rospy.Subscriber("sensors", Sensors, self.sensorsCallback)
+        self.pubWind = rospy.Publisher('windSensor', windSensor, queue_size=4)
+        self.pubGPS = rospy.Publisher('GPS', GPS, queue_size=4)
+
+        # sensorMsg = rospy.wait_for_message('sensors', Sensors)
+        self.wind_speedKmph = 0
+        self.wind_direction = 0
+        self.gps_lat = 0
+        self.gps_lon = 0
+        self.gps_headingDegrees = 0
+        self.gps_speedKmph = 0
+
+        self.past_wind_speedKmph = 0
+        self.past_wind_direction = 0
+        self.past_gps_lat = 0
+        self.past_gps_lon = 0
+        self.past_gps_headingDegrees = 0
+        self.past_gps_speedKmph = 0
+
+    def sensorsCallback(self, data):
+        self.data = data
+
+    def pub(self):
+        self.translate()
```

Comment:
Check if flag is set.
If so, then continue
Else, ros warn about not having received first sensor message yet.